src/script/cpp_api/s_client.cpp

```diff
@@ -94,7 +116,11 @@ void ScriptApiClient::on_hp_modification(int32_t newhp)
 	lua_getfield(L, -1, "registered_on_hp_modification");
 	// Call callbacks
 	lua_pushinteger(L, newhp);
-	runCallbacks(1, RUN_CALLBACKS_MODE_OR_SC);
+	try {
+		runCallbacks(1, RUN_CALLBACKS_MODE_OR_SC);
+	} catch (LuaError &e) {
+		getClient()->setFatalError(e);
+	}
```

Comment:
The client needs this because it does not have a central place to catch `LuaError`s